core/connection/tlsconfig.go

```diff
@@ -63,23 +75,56 @@ func (c *TLSConfig) AddClientCertFromFile(clientFile string) (string, error) {
 		return "", err
 	}
 
-	cert, err := tls.X509KeyPair(data, data)
+	var (
```

Comment:
Just as a preference, I tend to not use var blocks and instead declare each line with a var:

```
var currentBlock *pem.Block
var certBlock, certDecodeBlock, keyBlock []byte
```